mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -275,6 +278,63 @@ class DecisionTreeClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     val model = dt.fit(df)
   }
 
+  test("training with weighted data") {
+    val (dataset, weightedDataset) = {
+      val testData1 = TreeTests.generateNoisyData(5, 123)
+      val testData2 = TreeTests.generateNoisyData(5, 456)
+
+      // Over-sample the 1st dataset twice.
+      val overSampledTestData1 = testData1.flatMap {
+        labeledPoint => Iterator(labeledPoint, labeledPoint)
+      }
+
+      val rnd = new Random(8392)
+      val weightedTestData1 = testData1.flatMap {
```

Comment:
It is not obvious what the code is doing here. A comment might be useful to point out that you are weighting each sample by 2x, effectively, by using weighted?